managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/cache/RangeEntryCacheImpl.java

```diff
@@ -201,20 +201,6 @@ public void invalidateAllEntries(long ledgerId) {
     @Override
     public void asyncReadEntry(ReadHandle lh, PositionImpl position, final ReadEntryCallback callback,
             final Object ctx) {
-        try {
-            asyncReadEntry0(lh, position, callback, ctx);
-        } catch (Throwable t) {
-            log.warn("failed to read entries for {}-{}", lh.getId(), position, t);
```

Comment:
If `callback.readEntryComplete(cachedEntry, ctx);` throw any exception, the exception will also be catched by here, and invoke the `callback.readEntryFailed(createManagedLedgerException(t), ctx);`.

This means that in some cases we may call both methods, which is a risk to some recycle objects or other states.